search-app/src/cmr/search/services/content_service.clj

```diff
@@ -0,0 +1,153 @@
+(ns cmr.search.services.content-service
+  "Provides service functions for generating static content."
+  (:require
+   [clojure.string :as string]
+   [cmr.common-app.cache.consistent-cache :as consistent-cache]
+   [cmr.common-app.cache.cubby-cache :as cubby-cache]
+   [cmr.common.cache :as cache]
+   [cmr.common.cache.fallback-cache :as fallback-cache]
+   [cmr.common.cache.single-thread-lookup-cache :as stl-cache]
+   [cmr.common.config :refer [defconfig]]
+   [cmr.common.jobs :refer [defjob default-job-start-delay]]
+   [cmr.common.log :refer :all]
+   [cmr.common.util :as util]
+   [cmr.search.site.pages :as pages]
+   [cmr.search.site.static :as static]
+   [cmr.search.site.util :as site-utils]
+   [cmr.transmit.metadata-db :as mdb]))
+
+(def cache-key
+  "The key used to store the humanizer report cache in the system cache map."
+  :static-content-cache)
+
+(defconfig job-delay
+  "Number of seconds the `static-content-generator-job` needs to wait after
+  collection cache refresh job starts.
+
+  We need to add the delay so that the collection cache can be populated first
+  (collections are one of the key things queried when generating static
+  content)."
+  {:default 400
+   :type Long})
+
+(defconfig job-wait
+  "Number of milliseconds the `static-content-generator-job` waits for the
+  collection cache to be populated in the event when the delay is not long
+  enough."
+  {:default (* 60 1000)
+   :type Long})
+
+(defconfig retry-count
+  "Number of times `static-content-generator-job` retries to get the
+  collections from collection cache."
+  {:default 20
+   :type Long})
+
+(defconfig generation-interval
+  "Number of seconds between `static-content-generator-job` runs."
+  {;:default (* 60 60 24)
+   :default (* 60)
+   :type Long})
+
+(defn create-cache
+  "This function creates the composite cache that is used for caching the
+  static content data. With the given composition we get the following
+  features:
+  * A cubby cache that holds the generated content (centralized storage in
```

Comment:
What you are actually doing here is setting up a cache that will store a hash of the content in cubby. So all of the apps will know whenever the content changes and can trigger regathering the data.

Reply:
Are you talking about the internal mechanics? Or of the actual values stored?

My intent is that the actual values stored are the generated content, keyed off of the routes ...

Reply:
I was incorrect. You are storing the content in Elasticsearch. I confused it with the consistent cache which does what I described. Carry on.